src/future.cpp

```diff
@@ -145,17 +144,20 @@ CassError cass_future_custom_payload_item(CassFuture* future,
   }
   cass::SharedRefPtr<cass::Response> response(
         static_cast<cass::ResponseFuture*>(future->from())->response());
-  const cass::Response::CustomPayloadVec& custom_payload
-      = response->custom_payload();
-  if (index >= custom_payload.size()) {
-    return CASS_ERROR_LIB_INDEX_OUT_OF_BOUNDS;
+  if (response) {
+    const cass::Response::CustomPayloadVec& custom_payload =
+          response->custom_payload();
+    if (index >= custom_payload.size()) {
+      return CASS_ERROR_LIB_INDEX_OUT_OF_BOUNDS;
+    }
+    const cass::Response::CustomPayloadItem& item = custom_payload[index];
+    *name = item.name.data();
+    *name_length = item.name.size();
+    *value = reinterpret_cast<const cass_byte_t*>(item.value.data());
+    *value_size = item.value.size();
+    return CASS_OK;
   }
-  const cass::Response::CustomPayloadItem& item = custom_payload[index];
-  *name = item.name.data();
-  *name_length = item.name.size();
-  *value = reinterpret_cast<const cass_byte_t*>(item.value.data());
-  *value_size = item.value.size();
-  return CASS_OK;
+  return CASS_ERROR_LIB_INTERNAL_ERROR;
```

Comment:
What do you think about creating a new error for this?  I skimmed over the existing errors codes and couldn't find one that fits this type of error. I'm thinking something like 'CASS_ERROR_LIB_RESPONSE_NOT_AVAILABLE`.